app/Http/Middleware/MaintenanceMiddleware.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Pterodactyl\Http\Middleware;
+
+use Closure;
+
+class MaintenanceMiddleware
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        $server = $request->attributes->get('server');
```

Comment:
You probably just want to add a repository function to the `NodeRepository` class called `isInMaintenanceMode(int $nodeId): bool` and then you can just call `if ($repository->isInMaintenanceMode($server->node_id)` which then allows this to be unit tested better.

Currently the way you're doing this, $server->node will fire off a request to the database unless the node relationship is already loaded (which it generally is, but not the best assumption).

If you want to, you can do a `$server->relationLoaded('node')` call and if it is loaded check for `->maintenance`, otherwise fallback to the repository call.